packages/odf/components/topology/Topology.tsx

```diff
@@ -0,0 +1,512 @@
+import * as React from 'react';
+import HandleErrorAndLoading from '@odf/shared/error-handler/ErrorStateHandler';
+import { useDeepCompareMemoize } from '@odf/shared/hooks/deep-compare-memoize';
+import { DeploymentModel, NodeModel } from '@odf/shared/models';
+import { getName, getUID } from '@odf/shared/selectors';
+import {
+  createNode,
+  defaultLayoutFactory,
+  stylesComponentFactory,
+  TopologyDataContext,
+  TopologySearchContext,
+  TopologyViewLevel,
+} from '@odf/shared/topology';
+import {
+  DeploymentKind,
+  NodeKind,
+  PodKind,
+  StorageClusterKind,
+} from '@odf/shared/types';
+import {
+  K8sResourceCommon,
+  useK8sWatchResource,
+} from '@openshift-console/dynamic-plugin-sdk';
+import classNames from 'classnames';
+import * as _ from 'lodash-es';
+import {
+  useVisualizationController,
+  TopologyView,
+  VisualizationSurface,
+  Visualization,
+  VisualizationProvider,
+  SELECTION_EVENT,
+  TopologySideBar,
+  TopologyControlBar,
+  createTopologyControlButtons,
+  defaultControlButtonsOptions,
+  LabelPosition,
+  NodeShape,
+} from '@patternfly/react-topology';
+import { CEPH_STORAGE_LABEL } from '../../constants';
+import {
+  nodeResource,
+  odfDaemonSetResource,
+  odfDeploymentsResource,
+  odfPodsResource,
+  odfReplicaSetResource,
+  odfStatefulSetResource,
+  storageClusterResource,
+} from '../../resources';
+import { getZone } from '../../utils';
+import {
+  STEP_INTO_EVENT,
+  STEP_TO_CLUSTER,
+  ZOOM_IN,
+  ZOOM_OUT,
+} from './constants';
+import {
+  generateDeploymentsInNodes,
+  generateNodesInZone,
+  generateStorageClusterGroup,
+} from './NodeGenerator';
+import { TopologyTopBar } from './TopBar';
+import { generateNodeDeploymentsMap, groupNodesByZones } from './utils';
+import './topology.scss';
+
+const Sidebar: React.FC<{ selectedIds: string[]; onClose: any }> = ({
+  onClose,
+  selectedIds,
+}) => (
+  <TopologySideBar onClose={onClose} show={selectedIds.length > 0}>
+    Sidebar
+  </TopologySideBar>
+);
+
+const TopologyViewComponent: React.FC = React.memo(() => {
```

Comment:
`React.memo` only checks for prop changes. There are no props here. Is there any advantage of using it ??
Also, for `useContext` as well, FC will still re-render when context changes.
```suggestion
const TopologyViewComponent: React.FC = () => {
```

Reply:
yes you are right. removed